ecommerce-ios/Sources/Home/View/ProductCollectionView/ProductCollectionView.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  ProductCollectionView.swift
+//  ecommerce-ios
+//
+//  Created by Su T. Nguyen on 20/05/2021.
+//
+
+import SwiftUI
+
+struct ProductCollectionView: View {
+
+    private let viewModel = ProductCollectionViewViewModel()
+
+    private let numberOfColumns: Int = 2
+    private let spacing: CGFloat = 17.0
+
+    private var columns: [GridItem] {
+        let numberOfColumns = CGFloat(self.numberOfColumns)
+        let minimunWidth = ((screenWidth - spacing * (numberOfColumns + 1)) / numberOfColumns).rounded(.down)
+        return [.init(.adaptive(minimum: minimunWidth))]
+    }
+
+    var body: some View {
+        let sections = viewModel.sections.filter { $0.cellViewModels.count != 0 }
+
+        return LazyVStack {
+            ForEach(sections) { section in
+                VStack {
+                    HStack {
+                        Text(section.title)
+                            .font(.system(size: 22.0, weight: .bold))
+                        Spacer()
+                        Button("Shop all") {
+                            print("Did tap shop all button")
+                        }
+                        .font(.system(size: 15.0))
+                        .foregroundColor(.purpleBlue)
+                    }
```

Comment:
Please define a separated View for this. You can name it with ProductSectionHeaderView. I think we should make this one customizable with a title label and a trailing view. I suggest you to look at the Document how to customize view inside a view that I shared in Notion.